core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -374,6 +374,8 @@ void HIPInternal::finalize() {
   this->fence("Kokkos::HIPInternal::finalize: fence on finalization");
   was_finalized = true;
 
+  KOKKOS_IMPL_HIP_SAFE_CALL(hipSetDevice(m_hipDev));
```

Comment:
Same question here.  Why is that change necessary as part of this PR?

Reply:
This isn't necessary since `device_mem_space` is created with the appropriate device id.